frontend/src/metabase/meta/Parameter.js

```diff
@@ -67,7 +67,11 @@ export function getParameterTargetFieldId(
 ): ?FieldId {
   if (target && target[0] === "dimension") {
     const dimension = target[1];
-    if (Array.isArray(dimension) && dimension[0] === "template-tag") {
+    if (
+      Array.isArray(dimension) &&
+      dimension[0] === "template-tag" &&
+      datasetQuery !== undefined
```

Comment:
Generally if I expect an object I just check for falsey-ness, i.e. just `datasetQuery`, `!!datasetQuery`, or at most double equals like  `datasetQuery != null` which checks for `null` and `undefined` (really the only case you should ever use `==` in JS)